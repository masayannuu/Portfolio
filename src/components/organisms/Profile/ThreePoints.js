import React from 'react'
import styled from 'styled-components'
import { Image } from 'reactbulma'
import InViewMonitor from 'react-inview-monitor'

import SubHeading from '../../atoms/Title/SubHeading'
import StyledSection from '../../atoms/Section/StyledSection'
import Icon from '../../../../public/three-sibe-back.png'

const StyledDiv = styled.div`
  @media screen and (max-width: 768px) {
    width: 100%;
  }
  @media screen and (min-width: 769px) {
    width: 593px;
  }
  @media screen and (min-width: 1408px) {
    width: 100%;
  }
  font-feature-settings: 'palt';
  letter-spacing: 0.1em;
  -webkit-animation-duration: 2s;
  -moz-animation-duration: 2s;
  animation-duration: 2s;
  font-size: 1rem;
`

const StyledImage = styled(Image)`
  @media screen and (max-width: 768px) {
    width: 100%;
  }
  @media screen and (min-width: 769px) {
    position: -webkit-sticky;
    position: sticky;
    top: 50px;
    width: 367px;
    height: 593px;
  }
  @media screen and (min-width: 1408px) {
    position: -webkit-sticky;
    position: sticky;
    top: 50px;
    width: 50%;
    height: 593px;
  }
`

const StyledAnker = styled.a`
  color: #4a4a4a;
  &:hover {
    color: #a2c4c7;
  }
`

const StyledDl = styled.dl`
  margin-bottom: 1rem;
`

const ThreePoints = () => (
  <StyledSection id="ThreePoints">
    <SubHeading>3-CLUTURES</SubHeading>
    <div className="columns">
      <StyledDiv className="column">
        <InViewMonitor classNameNotInView="vis-hidden" classNameInView="animated slideInLeft">
          <StyledImage src={Icon} />
        </InViewMonitor>
      </StyledDiv>
      <StyledDiv className="column tk-a-otf-gothic-bbb-pr6n">
        <InViewMonitor classNameNotInView="vis-hidden" classNameInView="animated slideInUp">
          <StyledDl>
            <dt>
              <b>1. 毎日読書をしている</b>
            </dt>
            <dd>本や論文は好奇心と知識の源。行動と思考を補助してくれる叡智。</dd>
            <dd>
              漫画は
              <StyledAnker
                href="https://ja.wikipedia.org/wiki/%E4%B9%9D%E4%BA%95%E8%AB%92%E5%AD%90"
                target="_blank"
              >
                九井諒子先生
              </StyledAnker>
              の「
              <StyledAnker
                href="https://www.amazon.co.jp/dp/B00E59A03O/ref=dp-kindle-redirect?_encoding=UTF8&btkr=1"
                target="_blank"
              >
                ひきだしにテラリウム
              </StyledAnker>
              」が好き。
            </dd>
          </StyledDl>
          <StyledDl>
            <dt>
              <b>2. 構造思考で見つめる</b>
            </dt>
            <dd>システム思考フェチ。図解も好き。</dd>
            <dd>幾何学な設計と、それが寂びていく風景も好き。</dd>
          </StyledDl>
          <StyledDl>
            <dt>
              <b>3. スノーボードが好き</b>
            </dt>
            <dd>10歳離れた兄とのコミュニケーション手段から始まり、見事にハマった。</dd>
            <dd>バックカントリーで見た蔵王山の樹氷に心打たれている。</dd>
          </StyledDl>
        </InViewMonitor>
      </StyledDiv>
    </div>
  </StyledSection>
)

export default ThreePoints
