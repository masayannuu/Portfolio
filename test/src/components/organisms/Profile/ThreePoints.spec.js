import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import ThreePoints from '../../../../../src/components/Organisms/Profile/ThreePoints'

describe('ThreePoints ThreePoints Molecules', () => {
  it('should render ThreePoints', () => {
    const wrapper = Enzyme.shallow(<ThreePoints />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
