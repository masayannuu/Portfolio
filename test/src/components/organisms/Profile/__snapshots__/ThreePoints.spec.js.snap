// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreePoints ThreePoints Molecules should render ThreePoints 1`] = `
<StyledSection
  id="ThreePoints"
>
  <SubHeading>
    3-CLUTURES
  </SubHeading>
  <div
    className="columns"
  >
    <ThreePoints__StyledDiv
      className="column"
    >
      <InviewMonitor
        childPropsNotInView={Object {}}
        classNameInView="animated slideInLeft"
        classNameNotInView="vis-hidden"
        intoViewMargin="-20%"
        useInviewMonitor={[Function]}
      >
        <ThreePoints__StyledImage
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </InviewMonitor>
    </ThreePoints__StyledDiv>
    <ThreePoints__StyledDiv
      className="column tk-a-otf-gothic-bbb-pr6n"
    >
      <InviewMonitor
        childPropsNotInView={Object {}}
        classNameInView="animated slideInUp"
        classNameNotInView="vis-hidden"
        intoViewMargin="-20%"
        useInviewMonitor={[Function]}
      >
        <ThreePoints__StyledDl>
          <dt>
            <b>
              1. 毎日読書をしている
            </b>
          </dt>
          <dd>
            本や論文は好奇心と知識の源。行動と思考を補助してくれる叡智。
          </dd>
          <dd>
            漫画は
            <ThreePoints__StyledAnker
              href="https://ja.wikipedia.org/wiki/%E4%B9%9D%E4%BA%95%E8%AB%92%E5%AD%90"
              target="_blank"
            >
              九井諒子先生
            </ThreePoints__StyledAnker>
            の「
            <ThreePoints__StyledAnker
              href="https://www.amazon.co.jp/dp/B00E59A03O/ref=dp-kindle-redirect?_encoding=UTF8&btkr=1"
              target="_blank"
            >
              ひきだしにテラリウム
            </ThreePoints__StyledAnker>
            」が好き。
          </dd>
        </ThreePoints__StyledDl>
        <ThreePoints__StyledDl>
          <dt>
            <b>
              2. 構造思考で学んでいる
            </b>
          </dt>
          <dd>
            システム思考フェチ。図解も好き。
          </dd>
          <dd>
            幾何学な設計と、それが寂びていく風景も好き。
          </dd>
        </ThreePoints__StyledDl>
        <ThreePoints__StyledDl>
          <dt>
            <b>
              3. スノーボードが好き
            </b>
          </dt>
          <dd>
            10歳離れた兄とのコミュニケーション手段から始まり、見事にハマった。
          </dd>
          <dd>
            オフシーズンも運動は習慣化している。(健康的)
          </dd>
        </ThreePoints__StyledDl>
      </InviewMonitor>
    </ThreePoints__StyledDiv>
  </div>
</StyledSection>
`;
